packages/vscode-extension/src/common/Project.ts

```diff
@@ -101,6 +106,9 @@ export interface ProjectInterface {
   getDeviceSettings(): Promise<DeviceSettings>;
   updateDeviceSettings(deviceSettings: DeviceSettings): Promise<void>;
 
+  getWindowState(): Promise<WindowState>;
+  updateWindowState(newState: Partial<WindowState>): Promise<void>;
```

Comment:
I don't think we want to add this here. This is an interface for communicating between the frontend of the extension (what gets displayed in the panel) and the main extension process (what runs in NodeJS environment inside VSCode). It doesn't seem like it's needed by the main process to understand whether someone is hovering edges of the device.

The change you're making belongs fully into the frontend code. It's sole purpose is to present zoom controls in a way they don't cover device window and because of that there should be no changes made outside of the front-end part which lives under `packages/vscode-extension/src/webview/`